File: 1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	hash := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		if j, ok := hash[target-nums[i]]; ok {
			return []int{j, i}
		}
		hash[nums[i]] = i
	}
	return nil // 这里是为了编译通过，实际上不会执行到这里
}

// @lc code=end

File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python
#
# [1] 两数之和
#

# @lc code=start

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result = []
        hashmap = {}
        for i, num in enumerate(nums):
            hashmap[num] = i
        for i, num in enumerate(nums):
            diff = target-num
            j = hashmap.get(diff)
            if(j and j != i):
                return[i, j]


# @lc code=end
'''
心得：学会建立哈希表
使用enumerate的for循环
'''
